client.go

```diff
@@ -56,55 +64,45 @@ func NewSimpleClient(proxyAddr string) *Client {
 	}
 }
 
-// DialTCP when command is CONNECT and BIND,use tcp dail
-// return a TCPConn
-func (clt *Client) DialTCP(request *Request) *net.TCPConn {
-	if request.CMD == UDP_ASSOCIATE {
-		clt.logf()("command should not be udp_associate")
-		return nil
-	}
-	conn, err := clt.handShake(request)
+// TCPConnect socks TCP connect,get a tcp connect and reply addr
+func (clt *Client) TCPConnect(request *Request) (conn net.Conn, replyAddr *Address) {
+	proxyTCPConn, err := net.Dial("tcp", clt.ProxyAddr)
 	if err != nil {
 		clt.logf()(err.Error())
-		return nil
-	}
-	if tcpConn, ok := conn.(*net.TCPConn); !ok {
-		conn.Close()
-		clt.logf()("conn should be TCP conn")
-		return nil
-	} else {
-		return tcpConn
+		return nil, nil
 	}
-}
-
-// handShake socks protocol handshake process
-func (clt *Client) handShake(request *Request) (conn net.Conn, err error) {
-	proxyTCPConn, err := net.Dial("tcp", clt.ProxyAddr)
-	if err != nil {
-		return nil, err
+	if clt.TimeOut != 0 {
+		err = proxyTCPConn.SetDeadline(time.Now().Add(clt.TimeOut))
```

Comment:
After **handshake**, you should use net.Conn.SetReadline(time.Time{}) to cancel conn's timeout limit.